app/modules/github.py

```
@@ -0,0 +1,70 @@
+            raise Exception(f"failed to get assets from github release: {release_name}, response: {release_json}")
+        return assets
+
+    def _get_vm_json_link(self, release_name: str, assets: dict) -> str:
+        vm_json_link = next(
+            iter([x.get("browser_download_url", None) for x in assets if x.get("name", None) == "vm.json"]),
+            None,
+        )
+        if vm_json_link is None:
+            raise Exception(
+                f"failed to get download link from github release: {release_name}, response: {release_json}"
+            )
```

Ruff (0.13.3)

Create your own exception

(TRY002)